stock_intercompany/models/stock_picking.py

```diff
@@ -0,0 +1,27 @@
+from odoo import api, fields, models
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+
+    #override of method from stock module
+    def button_validate(self):
+        companies = self.env['res.company'].sudo().search([])
+        partners = {cp.partner_id: cp for cp in companies}
+
+        for picking in self:
+            if picking.partner_id in partners.keys():
+                company = partners[picking.partner_id]
+                warehouse = self.env["stock.warehouse"].sudo().search([('company_id','=',company.id)], limit=1)
+                vals = {
+                    'company_id': company.id,
+                    'picking_type_id': company.intercompany_in_type_id.id  or warehouse.in_type_id.id,
+                    'state':'draft',
+                    'location_id': self.env['stock.location'].sudo().search([('name','=','Vendors')]).id,
+                    'location_dest_id': self.env['stock.location'].sudo().search([('name','=','Stock'),('company_id','=',company.id)]).id,
+                }
+
+                new_picking_vals = picking.sudo().copy_data(default=vals)
+                reception_picking = self.env['stock.picking'].sudo().create(new_picking_vals)
+                reception_picking.action_confirm()
```

Comment:
Probably better to keep the reception picking in draft so the destination company user can handle in with more options.